File: apps/briefcase-mobile-signer/ios/BriefcaseMobileSigner/Core/BriefcasedModels.swift
import Foundation

struct SignerPairCompleteRequest: Codable {
    let msg1_b64: String
    let algorithm: String
    let signer_pubkey_b64: String
    let device_name: String?
}

struct SignerPairCompleteResponse: Codable {
    let msg2_b64: String
}

struct SignerSignedRequest: Codable {
    let signer_id: UUID
    let ts_rfc3339: String
    let nonce: String
    let sig_b64: String
}

struct ApprovalRequest: Codable, Identifiable, Hashable {
    let id: UUID
    let created_at: String
    let expires_at: String
    let tool_id: String
    let reason: String
    let summary: JSONValue
}

struct ListApprovalsResponse: Codable {
    let approvals: [ApprovalRequest]
}

struct ApproveResponse: Codable {
    let approval_id: UUID
    let approval_token: String
}

struct ErrorResponse: Codable, Error {
    let code: String
    let message: String
}

File: apps/briefcase-mobile-signer/android/app/src/main/java/com/credentialbriefcase/mobilesigner/BriefcasedModels.kt
package com.credentialbriefcase.mobilesigner

import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonElement

@Serializable
data class SignerPairCompleteRequest(
    val msg1_b64: String,
    val algorithm: String,
    val signer_pubkey_b64: String,
    val device_name: String? = null,
)

@Serializable
data class SignerPairCompleteResponse(
    val msg2_b64: String,
)

@Serializable
data class SignerSignedRequest(
    val signer_id: String,
    val ts_rfc3339: String,
    val nonce: String,
    val sig_b64: String,
)

@Serializable
data class ApprovalRequest(
    val id: String,
    val created_at: String,
    val expires_at: String,
    val tool_id: String,
    val reason: String,
    val summary: JsonElement,
)

@Serializable
data class ListApprovalsResponse(
    val approvals: List<ApprovalRequest>,
)

@Serializable
data class ApproveResponse(
    val approval_id: String,
    val approval_token: String,
)

@Serializable
data class ErrorResponse(
    val code: String,
    val message: String,
)

